File: 137-single-number-ii/single-number-ii.java
class Solution {
    public int singleNumber(int[] nums) {
        int ans = 0;
        for (int i = 0; i < 32; i++) {
            int sum = 0;
            for (int num : nums) {
                int val = num;
                sum += (num >> i) & 1;
            }
            sum %= 3;
            ans |= sum << i;
        }

        return ans;
    }
}

File: 137-single-number-ii/single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int res = 0;
        for (int i = 0; i <= 31; i++) {
            int sum = 0;
            for (int num : nums) {
                sum += (num >> i) & 1;
            }
            sum = sum % 3;
            res = res | (sum << i);
        }

        return res;
    }
};